gvm/protocols/gmpv7.py

```diff
@@ -1935,40 +1935,284 @@ def modify_filter(self, filter_id, comment=None, name=None, term=None,
-        return self.send_command(cmd)
+    def modify_group(self, group_id, comment=None, name=None,
+                     users=None):
+        """Generates xml string for modify group on gvmd.
 
-    def modify_note(self, note_id, text, **kwargs):
-        cmd = self._generator.modify_note_command(note_id, text, kwargs)
-        return self.send_command(cmd)
+        Arguments:
+            group_id (str): UUID of group to modify.
+            comment (str, optional): Comment on group.
+            name (str, optional): Name of group.
+            users (list, optional): List of user names to be in the group
+        """
+        if not group_id:
+            raise RequiredArgument('modify_group requires a group_id argument')
 
-    def modify_override(self, override_id, text, **kwargs):
-        cmd = self._generator.modify_override_command(override_id, text,
-                                                      kwargs)
-        return self.send_command(cmd)
+        cmd = XmlCommand('modify_group')
+        cmd.set_attribute('group_id', group_id)
 
-    def modify_permission(self, permission_id, **kwargs):
-        cmd = self._generator.modify_permission_command(
-            permission_id, kwargs)
-        return self.send_command(cmd)
+        if comment:
+            cmd.add_element('comment', comment)
 
-    def modify_port_list(self, port_list_id, **kwargs):
-        cmd = self._generator.modify_port_list_command(port_list_id, kwargs)
-        return self.send_command(cmd)
+        if name:
+            cmd.add_element('name', name)
+
+        if users:
+            cmd.add_element('users', ','.join(users))
+
+        return self._send_xml_command(cmd)
+
+    def modify_note(self, note_id, text, active=None, hosts=None, port=None,
+                    result_id=None, severity=None, task_id=None, threat=None):
+        """Generates xml string for modify note on gvmd.
+
+        Arguments:
+            note_id (str): UUID of note to modify.
+            text (str): The text of the note.
+            active (int, optional): Seconds note will be active.
+                -1 on always, 0 off.
+            hosts (list, optional): A list of hosts addresses
+            port (str, optional): Port to which note applies.
+            result_id (str, optional): Result to which note applies.
+            severity (str, optional): Severity to which note applies.
+            task_id (str, optional): Task to which note applies.
+            threat (str, optional): Threat level to which note applies.
+        """
+        if not note_id:
+            raise RequiredArgument('modify_note requires a note_id attribute')
+        if not text:
+            raise RequiredArgument('modify_note requires a text element')
+
+        cmd = XmlCommand('modify_note')
+        cmd.set_attribute('note_id', note_id)
+        cmd.add_element('text', text)
+
+        if not active is None:
+            cmd.add_element('active', '1' if active else '0')
```

Comment:
Oh I am sorry. The active parameter seems to be used differently in note. I thought it is a boolean but is an int. Your code was right last time sorry :cry:

Could we rename active to seconds_active then?
```suggestion
            cmd.add_element('active', seconds_active)
```